notebooks/classify/signature.py

```diff
@@ -0,0 +1,249 @@
+import numpy as np
+import pandas as pd
+import re
+import datetime
+import matplotlib.pyplot as plt
+import ipywidgets as widgets
+from typing import Dict, List
+
+class DataSignature:
+    def __init__(self, table: pd.DataFrame, DEBUG: bool = False):
+        """[Class for generating data signatures for a table and finding unique values within a signature.]
+
+        Args:
+            table (pd.DataFrame): [Pandas dataframe to geneate signatures for]
+            DEBUG (bool, optional): [Debug flag]. Defaults to False.
+        """
+        self.table = table
```

Comment:
I'm not super enthusiastic about storing the original DF in the instance.
One useful design tip is working out what data you really need to keep (and probably persist as a JSON or something) and what can be jettisoned.
If you think about how this class might be used:
1. Run it against all my tables in Sentinel
2. Create classes for each
3. Save the summarized data that gives me the useful bits of analysis/visualization capabililty
4. In future I can instantiate classes from the saved data and browse around/use it without having to
    run the whole analysis again or having to have massive amounts of raw data lying around